lib/internal/main/watch_mode.js

```diff
@@ -32,25 +32,29 @@ markBootstrapComplete();
 
 // TODO(MoLow): Make kill signal configurable
 const kKillSignal = 'SIGTERM';
-const kShouldFilterModules = getOptionValue('--watch-path').length === 0;
 const kWatchedPaths = ArrayPrototypeMap(getOptionValue('--watch-path'), (path) => resolve(path));
 const kPreserveOutput = getOptionValue('--watch-preserve-output');
+const kWatch = process.execArgv.some((arg) => arg === '--watch');
 const kCommand = ArrayPrototypeSlice(process.argv, 1);
 const kCommandStr = inspect(ArrayPrototypeJoin(kCommand, ' '));
 const args = ArrayPrototypeFilter(process.execArgv, (arg, i, arr) =>
-  arg !== '--watch-path' && arr[i - 1] !== '--watch-path' && arg !== '--watch' && arg !== '--watch-preserve-output');
+  !arg.startsWith('--watch-path') &&
```

Comment:
Why use `startsWith` instead of `!==`? How is it related to this change?
```suggestion
  !StringPrototypeStartsWith(arg, '--watch-path') &&
```

Reply:
From what I saw, in the test it is passing the arg like `--watch-path xxx` but in the doc I see `--watch-path=xxx` and in this case  `--watch-path=xxx` was not removed from the args passed to the child process. When that happend that caused an infite loop because the child was called like `node --watch-path=xxx entry.js`